### ISSUE TITLE
[BUG] [E][WiFiClient.cpp:335] setOption(): fail on fd -1, errno: 9, "Bad file number"

### ISSUE THREAD

**Issue by mbc99** [bertmelis/espMqttClient#32]
**Describe the bug**
When executing the tls-esp32 example and changing the ssl auth by the PSK auth I get an error that aparently is not critical since the code seems to work fine

Which platform, esp8266 or esp32? ESP32
Do you use TLS or not? PSK
Do you use an IDE (Arduino, Platformio...)? VSCode+PlatformIO
Which version of the Arduino framework? ` - framework-arduinoespressif32 @ 3.20004.220825 (2.0.4) `

Please include any debug output and/or decoded stack trace if applicable.
```
Connecting to MQTT...
[  1839][E][WiFiClient.cpp:335] setOption(): fail on fd -1, errno: 9, "Bad file number"
Connected to MQTT.
```

**Expected behaviour**
It works. However the error makes you wonder if something is wrong

**To Reproduce**
I will put the lines that I modified from the initial example to use PSK. Basically I changed the setup() function

**Example code**
```cpp
void setup() {
  Serial.begin(115200);
  Serial.println();
  Serial.println();

  WiFi.onEvent(WiFiEvent);

  //mqttClient.setInsecure();
  mqttClient.setPreSharedKey("testuser", MYSECRETPSK);
  //mqttClient.setCredentials(MQTT_USER, MQTT_PASS);
  mqttClient.onConnect(onMqttConnect);
  mqttClient.onDisconnect(onMqttDisconnect);
  mqttClient.onSubscribe(onMqttSubscribe);
  mqttClient.onUnsubscribe(onMqttUnsubscribe);
  mqttClient.onMessage(onMqttMessage);
  mqttClient.onPublish(onMqttPublish);
  mqttClient.setServer(MQTT_HOST, MQTT_PORT);
  mqttClient.setCleanSession(true);
```

**Comment by bertmelis** [comment]
See https://github.com/espressif/arduino-esp32/issues/7244

We suffer from this bug when trying to disable Nagle (`setNoDelay`) [here](https://github.com/bertmelis/espMqttClient/blob/main/src/Transport/ClientSecureSync.cpp#L20) and [here](https://github.com/bertmelis/espMqttClient/blob/main/src/Transport/ClientSecureSync.cpp#L26).

While this bug might slow down TCP communication it doesn't affect the overall working of the lib.

**Comment by bertmelis** [comment]
Will be fixed in #37 as soon as Platformio releases the 'new' Arduino core for ESP32.

**Comment by bertmelis** [comment]
Will be fixed in next release. Meanwhile, you can use the git version.

This requires Arduino for ESP32 version 2.0.5. Make sure you update before trying.
